engine/language_client_codegen/src/ruby/mod.rs

```diff
@@ -63,13 +63,19 @@ impl<'ir> TryFrom<(&'ir IntermediateRepr, &'ir crate::GeneratorArgs)> for RubyCl
                     .map(|c| {
                         let (_function, _impl_) = c.item;
                         let return_type = f.elem().output();
-                        let done = ir.distribute_metadata(&return_type).1.1.done;
-                        let state = ir.distribute_metadata(&return_type).1.1.state;
+                        let done = ir.distribute_metadata(&return_type).1 .1.done;
+                        let state = ir.distribute_metadata(&return_type).1 .1.state;
```

Comment:
```suggestion
                        let done = ir.distribute_metadata(&return_type).1.1.done;
                        let state = ir.distribute_metadata(&return_type).1.1.state;
```

I had no idea you can put a space between a record and a field accessor! Your code parses correctly, but maybe it's less surprising without the space.